cloudsync/api_test.py

```diff
@@ -130,3 +150,148 @@ def test_process_transcode_results(mocker):
+    for item in filenames:
+        assert extract_title(item[0]) == item[1]
+
+
+def test_extract_bad_title():
+    """Assert that a VideoFilenameError is raised for a bad filename"""
+    bad_names = ('MIT-6.046-1510-L01.mp4',  # Missing -lec-mit-0000
+                 'Completely random name.mp4',  # No similarity to expected format
+                 'MIT-6.046-lec-mit-0000-2017apr06-L01')  # no extension
+
+    for name in bad_names:
+        with pytest.raises(VideoFilenameError):
+            extract_title(name)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_nouser():
+    """
+    Test that an exception is correctly handled
+    """
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with pytest.raises(User.DoesNotExist) as exc:
+        process_watch_file(upload)
+        assert 'User matching query does not exist' in str(exc)
+    with pytest.raises(Video.DoesNotExist):
+        Video.objects.get(title=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_s3_error():
+    """Test that an AWS S3 ClientError is correctly handled"""
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with transaction.atomic():
+        with pytest.raises(ClientError) as exc:
+            process_watch_file(upload)
+            assert 'NoSuchBucket' in str(exc)
+    with pytest.raises(Video.DoesNotExist):
+        Video.objects.get(title=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_filename_error():
+    """Test that a bad filename raises a VideoFilenameError"""
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'Bad filename.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with transaction.atomic():
+        with pytest.raises(VideoFilenameError) as exc:
+            process_watch_file(upload)
+            assert 'No matches found for filename' in str(exc)
+    with pytest.raises(Video.DoesNotExist):
+        Video.objects.get(title=upload)
```

Comment:
nit: you can also simply `assert not Video.objects.filter(title=upload).exists()` and not dealing with the exception